manifest/schema2/manifest.go

```diff
@@ -48,6 +52,14 @@ func init() {
 	}
 }
 
+// LayerDescriptor is an extension to distribution.Descriptor that
+// has additional fields specific to layers.
+type LayerDescriptor struct {
```

Comment:
For the most part, `URLs` can be safely ignored. It is only when the mediatype is encountered that a url entry should be used, since that resource may only be realized by a url from that set. Make sure to add `omitempty` on the field declaration so it is not included when not specified.